### PR TITLE
Adding test case - Sharing backup with multiple users and groups with different access levels

### PR DESCRIPTION
**Pull request by mkoppal-px**
**What this PR does / why we need it**:
This PR addresses the following:

- Added the automated test case for Sharing backup with multiple users and groups with different access levels
- Added helper function for getting context of a user by passing username and password - `GetNonAdminCtx `
- Added helper function to get a random user from a group - `GetRandomUserFromGroup`
- Made sure that `CreateSourceAndDestClusters`, `DeleteCluster` and `DeleteRestore` functions now accept the user context as input instead of always using admin context
- Removed time.Sleep from `CreateRestore` and used `task.DoRetryWithTimeout`

**Which issue(s) this PR fixes** (optional)
Closes #PA-314

**Special notes for your reviewer**:

### CODE REVIEW

File: tests/backup/backup_test.go [hunk 1]
```diff
@@ -611,52 +630,217 @@ var _ = Describe("{ShareBackupWithUsersAndGroups}", func() {
...
+			log.Infof("About to delete restore - %s", restoreName)
+			DeleteRestore(restoreName, orgID, ctxNonAdmin)
+
+			// Get Backup UID
+			backupDriver := Inst().Backup
+			backupUID, err := backupDriver.GetBackupUID(ctx, backupNames[0], orgID)
+			log.FailOnError(err, "Failed while trying to get backup UID for - %s", backupNames[0])
+
+			// Delete backup to confirm that the user has Full Access
+			backupDeleteResponse, err := DeleteBackup(backupNames[0], backupUID, orgID, ctxNonAdmin)
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:687]: add a check for err also here

**Review by snigdha-px — APPROVED** [review]
LGTM

File: tests/backup/backup_test.go [hunk 2]
```diff
@@ -3013,16 +3248,30 @@ func CreateRestore(restoreName string, backupName string,
...
 	}
-	time.Sleep(time.Minute * 3)
-	resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
-	log.FailOnError(err, "Verifying restore")
-	dash.VerifyFatal(resp.GetRestore().GetStatus().Status, api.RestoreInfo_StatusInfo_PartialSuccess, "Verifying restore")
+
+	restoreSuccessCheck := func() (interface{}, bool, error) {
+		resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+		log.FailOnError(err, "Failed verifying restore for - %s", restoreName)
+		if resp.GetRestore().GetStatus().GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess || resp.GetRestore().GetStatus().GetStatus() == api.RestoreInfo_StatusInfo_Success {
```
Comment (lsrinivas-pure) [inline tests/backup/backup_test.go:3263]: since you are using resp.GetRestore().GetStatus() mulitple time, can you put it in a varible name and resuse it?

**Review by lsrinivas-pure — COMMENTED** [review]
have a minor comments

File: tests/backup/backup_test.go (hunk 1 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go:687]: Done. Added log.FailOnError

File: tests/backup/backup_test.go (hunk 2 above)
Comment (mkoppal-px) [reply tests/backup/backup_test.go]: Done

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**